beacon-chain/core/epoch/precompute/justification_finalization.go

```
@@ -122,15 +128,23 @@ func weighJustificationAndFinalization(state state.BeaconState,
 		if err := state.SetCurrentJustifiedCheckpoint(&ethpb.Checkpoint{Epoch: currentEpoch, Root: blockRoot}); err != nil {
 			return nil, err
 		}
-		newBits := state.JustificationBits()
+		newBits, err := state.JustificationBits()
+		if err != nil {
+			return nil, err
+		}
 		newBits.SetBitAt(0, true)
 		if err := state.SetJustificationBits(newBits); err != nil {
 			return nil, err
 		}
 	}
 
 	// Process finalization according to Ethereum Beacon Chain specification.
-	justification := state.JustificationBits().Bytes()[0]
+	justificationBits, err := state.JustificationBits()
```

This shouldnt need to return an error, it is valid in all 3 states